src/backend/collectibles_types.nim

```diff
@@ -191,209 +143,145 @@ proc fromJson*(t: JsonNode, T: typedesc[ref CollectibleTrait]): ref CollectibleT
-    imageUrl:{self.imageUrl},
-    traits:{self.traits}
+    imageUrl:{self.imageUrl}
   )"""
 
-proc getCollectionTraits*(t: JsonNode): Table[string, CollectionTrait] =
-  var traitList: Table[string, CollectionTrait] = initTable[string, CollectionTrait]()
-  for key, value in t{"traits"}.getFields():
-    traitList[key] = fromJson(value, CollectionTrait)
-  return traitList
-
-proc fromJson*(t: JsonNode, T: typedesc[CollectionData]): CollectionData {.inline.} =
+proc fromJson*(t: JsonNode, T: typedesc[CollectionData]): CollectionData =
   result = CollectionData()
   result.name = t["name"].getStr()
   result.slug = t["slug"].getStr()
   result.imageUrl = t["image_url"].getStr()
-  result.traits = getCollectionTraits(t["traits"])
 
-proc fromJson*(t: JsonNode, T: typedesc[ref CollectionData]): ref CollectionData {.inline.} =
+proc fromJson*(t: JsonNode, T: typedesc[ref CollectionData]): ref CollectionData =
   result = new(CollectionData)
   result[] = fromJson(t, CollectionData)
 
 # CollectibleData
 proc `$`*(self: CollectibleData): string =
   return fmt"""CollectibleData(
-    id:{self.id},
-    communityId:{self.communityId},
     name:{self.name},
     description:{self.description},
-    permalink:{self.permalink},
     imageUrl:{self.imageUrl},
     animationUrl:{self.animationUrl},
     animationMediaType:{self.animationMediaType},
     traits:{self.traits},
-    backgroundColor:{self.backgroundColor},
-    tokenUri:{self.tokenUri},
+    backgroundColor:{self.backgroundColor}
   )"""
 
-proc getCollectibleTraits*(t: JsonNode): seq[CollectibleTrait] =
+proc getCollectibleTraits*(t: JsonNode): Option[seq[CollectibleTrait]] =
   var traitList: seq[CollectibleTrait] = @[]
   for item in t.getElems():
     traitList.add(fromJson(item, CollectibleTrait))
-  return traitList
+  if traitList.len == 0:
+    return none(seq[CollectibleTrait])
+  else:
+    return some(traitList)
 
-proc fromJson*(t: JsonNode, T: typedesc[CollectibleData]): CollectibleData {.inline.} =
+proc fromJson*(t: JsonNode, T: typedesc[CollectibleData]): CollectibleData =
   result = CollectibleData()
-  result.id = fromJson(t["id"], CollectibleUniqueID)
-  result.communityId = t["community_id"].getStr()
+
   result.name = t["name"].getStr()
-  result.description = t["description"].getStr()
-  result.permalink = t["permalink"].getStr()
-  result.imageUrl = t["image_url"].getStr()
-  result.animationUrl = t["animation_url"].getStr()
-  result.animationMediaType = t["animation_media_type"].getStr()
-  result.traits = getCollectibleTraits(t["traits"])
-  result.backgroundColor = t["background_color"].getStr()
-  result.tokenUri = t["token_uri"].getStr()
+  let descriptionNode = t{"description"}
+  if descriptionNode != nil and descriptionNode.kind != JNull:
+    result.description = some(descriptionNode.getStr())
+  else:
+    result.description = none(string)
+  let imageUrlNode = t{"image_url"}
+  if imageUrlNode != nil and imageUrlNode.kind != JNull:
```

Comment:
I spent stupid amounts of time trying to make a generic `Option` unmarshaller like we have in Activity, but apparently for whatever reason Nim didn't handle recursively calling generic procs with different types, I ended up getting a runtime crash...